andela/boondocks-bn-frontend
# #169257400 Logout Functionality

#### What does this PR do?
This PR creates a logout component that allows users to log out of the application

#### Description of Task to be completed?
- [x] create a logout component
- [x] add the component to our routes
- [x] add the logout link to our Navbar

#### How should this be manually tested?
- clone this repo and add the following variable to your './.env' file:
`API_URL=https://boondocks-bn-backend-staging.herokuapp.com`
- navigate to the project folder using Terminal
- install dependencies using `npm install`
- run tests using `npm test`
- start the App using `npm run watch` and it will be rendered on your browser
- log in to the application
- using developer tools, run the following commands to see the generated cookie and localStorage objects: `document.cookie` and `localStorage.bn_user_data` respectively
- use the logout link to log out of the application
- again, using developer tools, run the following commands to see the generated cookie and localStorage objects `document.cookie` and `localStorage.bn_user_data` respectively, they will have been cleared from the browser

#### Any background context you want to provide?
The Navbar is currently not re-rendering on user login and logout so a bug fix will be created for that. Currently, we must refresh the page manually to get the correct Navbar after login/logout.

#### What are the relevant pivotal tracker stories?
[#169257400](https://www.pivotaltracker.com/story/show/169257400)

#### Screenshots (if appropriate)
<img width="1440" alt="Screen Shot 2020-01-10 at 03 00 04" src="https://user-images.githubusercontent.com/17108099/72117205-7e27dd80-3355-11ea-9415-d7ea7ff8ccd0.png">

#### Questions:
N/A
